File: boj/15650.py
import sys
input = sys.stdin.readline

n, m = list(map(int, input().split()))
stack = []


def dfs(start):
    if len(stack) == m:
        print(' '.join(list(map(str, stack))))
        return

    for i in range(start, n+1):
        if i not in stack:
            stack.append(i)
            dfs(i+1)
            stack.pop()


dfs(1)

File: boj/15650.js
function nAndM(n, m, start, trace) {
  if (trace.length === m) {
    console.log(trace.join(' '))
    return
  }
  for (let i = start; i < n + 1; i++) {
    trace.push(i)
    nAndM(n, m, i + 1, trace)
    trace.pop()
  }
}

nAndM(4, 2, 1, [])
